### PR TITLE
comp : home page,header,footer,cat-card,quiz-card

### PR DESCRIPTION
**Pull request by manojsarna**
HomePage Added

Header Added

Footer Added

Category Cards Added

Quiz Cards Added

Feature : dark Mode Added , Site Responsiveness with Media Queries

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "liveServer.settings.port": 5510
+}
```
Comment (PuneetP16) [inline .vscode/settings.json:3]: Add this to git ignore.

File: lib/Components/Header/header.css [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    color: var(--COLOR-HEADER-FONT-DARK-HOVER);
+}
+.color-dm {
+    color: var(--COLOR-HEADER-FONT-DARK);
+}
+
+.sm-action-icon-btn-wrapper .sm-icon-btn-primary:hover {
+    color: var(--COLOR-HEADER-FONT-DARK-HOVER);
+}
+
```
Comment (abhishektanwar) [inline lib/Components/Header/header.css]: Please try to remain consistent with spacing between css classes.

File: lib/Components/Header/header.css [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#sc-login{
+    white-space: nowrap;
+}
+
+#sc-logo {
+    white-space: nowrap;
+    display: inline-block;
+}
```
Comment (abhishektanwar) [inline lib/Components/Header/header.css]: try not to use id for styling

File: lib/Components/Header/header.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const headerLoginBtn = document.getElementById("sc-login");
+
+headerLoginBtn.addEventListener("click", () => {
+  window.location = "/lib/components/auth/auth.html";
+});
```
Comment (abhishektanwar) [inline lib/Components/Header/header.js:5]: Why are we using this way of changing URL and not directly changing URL through href in anchor tag?

File: lib/Components/Header/header.css [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+/* HEADER ENDS */
+
+/* MEDIA QUERIES FOR HEADERS*/
+@media only screen and (max-width:768px){
+    :root{
+        font-size:45%;
+    }
+    .sm-nav-search{
```
Comment (hsnice16) [inline lib/Components/Header/header.css]: Nice work, but I will suggest to give at least a line space between two declarations.
```css
:root {
}

.sm-nav-search {
}

```

File: lib/Components/QuizCard/quizcard.css [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+.sm-quiz-card-title:hover {
+    background: var(--COLOR-BLACK-10);
+}
+
+.sm-quiz-card-img img{
+    width: 100%;
+    height: 25rem;
+    object-fit: cover;
+    border-radius: 30px;
```
Comment (abhishektanwar) [inline lib/Components/QuizCard/quizcard.css]: Maybe use rems here?

**Review by hsnice16 — APPROVED** [review]

File: lib/Components/QuizCard/quizcard.css [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    height: 100%;
+    bottom: 0;
+    right: 0;
+    top: 0;
+    left: 0;
+    overflow: hidden;
+    color: var(--COLOR-WHITE-ALWAYS);
+    text-shadow: 0px 5px 5px var(--COLOR-BLACK);
+    font-weight: 800;
+    border-radius: 30px;
```
Comment (abhishektanwar) [inline lib/Components/QuizCard/quizcard.css]: Maybe use rems here?

File: lib/Components/Header/header.css [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+#sc-logo {
+    white-space: nowrap;
+    display: inline-block;
+}
+
+.btn-social{
+    outline: none;
+    border:none;
+    background: none;
```
Comment (PuneetP16) [inline lib/Components/Header/header.css]: Suggestion: Make consistent use of selectors, if you are using everywhere classes, use classes.

**Review by abhishektanwar — APPROVED** [review]
UI looks great .Clean and readable code.Some suggestions have been commented,  Good to merge.

File: style.css [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+    bottom: 0;
+    right: 0;
+    top: 0;
+    left: 0;
+    overflow: hidden;
+    color: var(--COLOR-WHITE-ALWAYS);
+    text-shadow: 0px 5px 5px var(--COLOR-BLACK);
+    font-weight: 800;
+    border-radius: 30px;
+    background: var(--COLOR-BLACK-40);
```
Comment (PuneetP16) [inline style.css]: Suggestion: Use rems for making it responsive with respect to root font size.

File: lib/Components/QuizCard/quizcard.css [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+
+.sm-quiz-card-container:hover {
+    transform: scale(1.01);
+}
+
+
+.sm-quiz-card-img{
+    position: relative;
```
Comment (PuneetP16) [inline lib/Components/QuizCard/quizcard.css:25]: could remove unnecessary spaces.

**Review by PuneetP16 — APPROVED** [review]
Liked your implementation of dynamic images on featured categories. Good to merge after necessary changes.

File: lib/Components/Header/header.js (hunk 4 above)
Comment (bhtibrewal) [reply lib/Components/Header/header.js:5]: +1

File: lib/Components/Header/header.css (hunk 3 above)
Comment (manojsarna) [reply lib/Components/Header/header.css]: Changed

File: lib/Components/Header/header.js (hunk 4 above)
Comment (manojsarna) [reply lib/Components/Header/header.js:5]: yeah it can be done through hrefs also , just working out through buttons

**Review by yash9263 — APPROVED** [review]

File: lib/Components/QuizCard/quizcard.css (hunk 6 above)
Comment (manojsarna) [reply lib/Components/QuizCard/quizcard.css]: okay

File: .vscode/settings.json (hunk 1 above)
Comment (manojsarna) [reply .vscode/settings.json:3]: okay

File: lib/Components/Header/header.css (hunk 8 above)
Comment (manojsarna) [reply lib/Components/Header/header.css]: okay

**Pull request merged**